src/Modules/Reports/Commands.Reports/GetPowerBIReport.cs

```diff
@@ -3,31 +3,146 @@
+        [Parameter(Mandatory = false, ParameterSetName = ObjectListParameterSetName)]
+        public string Filter { get; set; }
+
+        [Alias("Top")]
+        [Parameter(Mandatory = false, ParameterSetName = ListParameterSetName)]
+        [Parameter(Mandatory = false, ParameterSetName = ObjectListParameterSetName)]
+        public int? First { get; set; }
+
+        [Parameter(Mandatory = false, ParameterSetName = ListParameterSetName)]
+        [Parameter(Mandatory = false, ParameterSetName = ObjectListParameterSetName)]
+        public int? Skip { get; set; }
+
+        [Alias("GroupId")]
+        [Parameter(Mandatory = false, ParameterSetName = IdParameterSetName)]
+        [Parameter(Mandatory = false, ParameterSetName = NameParameterSetName)]
+        [Parameter(Mandatory = false, ParameterSetName = ListParameterSetName)]
+        public Guid WorkspaceId { get; set; }
+
+        [Alias("Group")]
+        [Parameter(Mandatory = true, ParameterSetName = ObjectIdParameterSetName, ValueFromPipeline = true)]
+        [Parameter(Mandatory = true, ParameterSetName = ObjectNameParameterSetName, ValueFromPipeline = true)]
+        [Parameter(Mandatory = true, ParameterSetName = ObjectListParameterSetName, ValueFromPipeline = true)]
+        public Workspace Workspace { get; set; }
+        #endregion
+
+        #region Constructors
         public GetPowerBIReport() : base() { }
 
         public GetPowerBIReport(IPowerBIClientCmdletInitFactory init) : base(init) { }
+        #endregion
+
+        protected override void BeginProcessing()
+        {
+            base.BeginProcessing();
+            if(this.Scope == PowerBIUserScope.Individual && !string.IsNullOrEmpty(this.Filter))
+            {
+                this.Logger.ThrowTerminatingError($"{nameof(this.Filter)} is only applied when -{nameof(this.Scope)} is set to {nameof(PowerBIUserScope.Organization)}", ErrorCategory.InvalidArgument);
+            }
+        }
 
         public override void ExecuteCmdlet()
         {
+            if(this.Workspace != null)
+            {
+                this.WorkspaceId = this.Workspace.Id;
+            }
+
+            if (this.ParameterSet.Equals(IdParameterSetName))
+            {
+                this.Filter = $"id eq '{this.Id}'";
+            }
+
+            if (this.ParameterSet.Equals(NameParameterSetName))
+            {
+                this.Filter = $"tolower(name) eq '{this.Name.ToLower()}'";
+            }
+
+            IEnumerable<Report> reports = null;
             using (var client = this.CreateClient())
             {
-                var reports = client.Reports.GetReports();
-                this.Logger.WriteObject(reports, true);
+                if(this.WorkspaceId != default)
+                {
+                    reports = this.Scope == PowerBIUserScope.Individual ?
+                        client.Reports.GetReportsForWorkspace(this.WorkspaceId) :
```

Comment:
Functionally this is fine and makes the cmdlet very powerful and versatile. We could consider separating out the workspace part into its own cmdlet, but this is probably overkill. I'll only comment on this cmdlet, but it could be applicable to the others.

Reply:
I debated this too, I did it in the same cmdlet for simplicity to the end user